src/main/java/mat/server/CQLConstantServiceImpl.java

```diff
@@ -205,4 +220,28 @@ private CQLTypeContainer getCQLTypeInformation() {
 	private QDMContainer getQDMInformation() {
 		return QDMUtil.getQDMContainer();
 	}
+
+	public void getFhirAttributeAndDataTypes(CQLConstantContainer cqlConstantContainer) {
+		List<String> fhirAttributeList = new ArrayList<>();
+		List<String> fhirDataTypeList = new ArrayList<>();
+
+		try {
+			JSONArray fhirResponse = new JSONArray(restTemplate.getForObject(fhirMatServicesUrl + fhirMatServiceResouceForAttributes, String.class));
```

Comment:
I would suggest to get rid of json library usage if possible.
Did you try spring template convenience methods for that? Like getForEntity (..... String[].class) ?
If it works, it will result in cleaner code.

Reply:
it's not string[].class, we can use Pojo/Jackson to map an entry type

Reply:
Working on it